mlir/lib/Dialect/MIOpen/Transforms/AlignTiling.cpp

```diff
@@ -522,15 +667,96 @@ template <typename T> struct MILARewritePattern : public OpRewritePattern<T> {
   }
 };
 
-//===- Passes -------------------------------------------------------------===//
-//===- MIOpenLinalgAlignPass - Align Tiling of Linalg Ops -----------------===//
+//===----------------------------------------------------------------------===//
+// ThreadwiseCopyV2 Conversion in Fusion.
+//===----------------------------------------------------------------------===//
+
+struct ThreadwiseCopyV2RewritePattern
+    : public OpRewritePattern<miopen::ThreadwiseCopyV2Op> {
```

Comment:
You should have the ThreadwiseCopyV2 already in the previous pass, is there some missing info that prevents this from applying then. I would prefer a function instead of retracing the whole graph.